src/backend/distributed/operations/repair_shards.c

```diff
@@ -421,6 +423,48 @@ EnsureEnoughDiskSpaceForShardMove(List *colocatedShardList,
 }
 
 
+/*
+ * ErrorIfTargetNodeIsNotSafeToMove throws error if the target node is not
+ * eligible for moving shards.
+ */
+void
+ErrorIfTargetNodeIsNotSafeToMove(const char *targetNodeName, int targetNodePort)
+{
+	WorkerNode *workerNode = FindWorkerNode(targetNodeName, targetNodePort);
+	if (workerNode == NULL)
+	{
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("Moving shards to a non-existing node is not supported"),
+						errhint(
+							"Add the target node via SELECT citus_add_node('%s', %d);",
```

Comment:
Minor: Should we separate the commands with `:` so instead of `via SELECT ..` -> `via: SELECT ..`?